etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/TransformationRule.java

```diff
@@ -0,0 +1,79 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.Rule;
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The TransformationRule class extends the {@link
+ * gov.hhs.cdc.trustedintermediary.etor.ruleengine.Rule Rule} class and represents a transformation
+ * rule. It implements the {@link
+ * gov.hhs.cdc.trustedintermediary.etor.ruleengine.Rule#runRule(FhirResource) runRule} method to
+ * apply a transformation to the FHIR resource.
+ */
+public class TransformationRule extends Rule<TransformationRuleMethod> {
+
+    /**
+     * Do not delete this constructor! It is used for JSON deserialization when loading rules from a
+     * file.
+     */
+    public TransformationRule() {}
+
+    public TransformationRule(
+            String ruleName,
+            String ruleDescription,
+            String ruleMessage,
+            List<String> ruleConditions,
+            List<TransformationRuleMethod> ruleActions) {
+        super(ruleName, ruleDescription, ruleMessage, ruleConditions, ruleActions);
+    }
+
+    public void runRule(FhirResource<?> resource) {
+
+        for (TransformationRuleMethod transformation : this.getRules()) {
+            String name = transformation.name();
+            Map<String, String> args = transformation.args();
+
+            try {
+                Class<?> clazz = loadCustomTransformationClassFromFile(name);
+                Method method = clazz.getDeclaredMethod("transform", FhirResource.class, Map.class);
+                method.invoke(clazz.getDeclaredConstructor().newInstance(), resource, args);
+            } catch (ClassNotFoundException
+                    | NoSuchMethodException
+                    | IllegalAccessException
+                    | InvocationTargetException
+                    | InstantiationException e) {
+                logger.logError("Error invoking method: " + name, e);
+            }
+        }
+    }
+
+    private static Class<?> loadCustomTransformationClassFromFile(String className)
+            throws ClassNotFoundException {
+        String customPackageName =
+                "gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom";
+        Path customTransformationPath =
+                Paths.get(
+                        System.getProperty("user.dir"),
```

Comment:
I know we use the ApplicationContext for getting environment variables. I wonder if it would work in here as well?

Reply:
Good point. I think we need a standard way to easily construct paths. ApplicationContext seems like a good option